DuckDuckGo/NavigationBar/View/AddressBarTextField.swift

```diff
@@ -1098,6 +1103,16 @@ extension AddressBarTextField: NSTextViewDelegate {
         isUndoDisabled = false
     }
 
+    func textView(_ textView: NSTextView, shouldChangeTextIn affectedCharRange: NSRange, replacementString: String?) -> Bool {
+        // When a suggestion is visible and the user presses Delete, the suggestion should get removed but the text should remain as-is.
+        if self.window?.currentEvent?.specialKey == NSEvent.SpecialKey.delete, lastValueChangeRemovedSuggestion {
+            lastValueChangeRemovedSuggestion = false
+            return false
+        }
+
+        return true
+    }
```

Comment:
I would have preferred this function to check whether a suggestion was visible and instead remove it from the `self.value`, allowing deletion when no suggestion was visible, but the order of delegate events makes that tricky as `handleTextDidChange` gets called _before_ this one.